image/tarexport/save.go

```diff
@@ -174,15 +174,13 @@ func (s *saveSession) save(outStream io.Writer) error {
 		reposFile := filepath.Join(tempDir, legacyRepositoriesFileName)
 		f, err := os.OpenFile(reposFile, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644)
 		if err != nil {
-			f.Close()
 			return err
 		}
+		defer f.Close()
```

Comment:
It might be important to close this file before calling `Tar` below, so I'm not sure we should use `defer` here.